yb-voyager/cmd/assessMigrationCommandBulk.go

```diff
@@ -0,0 +1,316 @@
+	assessMigrationBulkCmd.Flags().Bool("ignore-exists", true, "If true, skip assessment if for a schema it's already done")
+	assessMigrationBulkCmd.Flags().StringVar(&bulkAssessmentDir, "bulk-assessment-dir", "", "Top-level directory storing the export-dir of each schema (default: pwd)")
+	BoolVar(assessMigrationBulkCmd.Flags(), &startClean, "start-clean", false, "Cleans up all the export-dirs in bulk assessment directory to start everything from scratch")
+
+	// marking mandatory flags
+	assessMigrationBulkCmd.MarkFlagRequired("fleet-config-file")
+	assessMigrationBulkCmd.MarkFlagRequired("bulk-assessment-dir")
+}
+
+func assessMigrationBulk() {
+	if startClean {
+		proceed := utils.AskPrompt(
+			"CAUTION: Using --start-clean will delete all progress in each export directory present inside the bulk-assessment-dir. " +
+				"Do you want to proceed")
+		if !proceed {
+			return
+		}
+
+		// cleaning all export-dir present inside bulk-assessment-dir
+		matches, err := filepath.Glob(fmt.Sprintf("%s/*-export-dir", bulkAssessmentDir))
+		if err != nil {
+			utils.ErrExit("error while cleaning up export directories: %s", err)
+		}
+		for _, match := range matches {
+			utils.CleanDir(match)
+		}
+
+		err = os.RemoveAll(filepath.Join(bulkAssessmentDir, "bulkAssessmentReport.html"))
+		if err != nil {
+			utils.ErrExit("failed to remove bulk assessment report: %s", err)
+		}
+	}
+
+	dbConfigs, err := parseFleetConfigFile(fleetConfigPath)
+	if err != nil {
+		utils.ErrExit("failed to parse fleet config file: %v", err)
+	}
+
+	for _, dbConfig := range dbConfigs {
+		utils.PrintAndLog("\nAssessing '%s' schema\n", dbConfig.Schema)
+		err = executeAssessment(dbConfig)
+		if err != nil {
+			log.Errorf("failed to assess migration for schema %s: %v", dbConfig.Schema, err)
+			fmt.Printf("failed to assess migration for schema %s: %v\n", dbConfig.Schema, err)
+			if !continueOnError {
+				break
+			}
+		}
+	}
+
+	err = generateBulkAssessmentReport(dbConfigs)
+	if err != nil {
+		utils.ErrExit("failed to generate bulk assessment report: %s", err)
+	}
+}
+
+func executeAssessment(dbConfig AssessMigrationDBConfig) error {
+	log.Infof("executing assessment for schema %q", dbConfig.Schema)
+	exportDirPath := getAssessmentExportDirPath(dbConfig)
+	cmdArgs := buildCommandArguments(dbConfig, exportDirPath)
+	if err := os.MkdirAll(exportDirPath, 0755); err != nil {
+		return fmt.Errorf("creating export-directory %q for schema %q: %w", exportDirPath, dbConfig.Schema, err)
+	}
+
+	execCmd := exec.Command(os.Args[0], cmdArgs...)
+	// password either has to be provided via fleet_config_file or can be provided at run-time by the user.
+	// user setting the env var route is not supported for assess-migration-bulk command
+	execCmd.Env = append(os.Environ(), "SOURCE_DB_PASSWORD="+dbConfig.Password)
+	execCmd.Stdout = os.Stdout
+	execCmd.Stderr = os.Stderr
+	execCmd.Stdin = os.Stdin
+	log.Infof("executing the cmd: %s", execCmd.String())
+	err := execCmd.Run()
```

Comment:
This should already be implemented (via the atexit handlers), but pls test and ensure that if we ctrl-c interrupt/ SIGTERM the main voyager process, the child process is also shutdown.

Reply:
yeah let me verify

Reply:
Overall, let's test the following cases:
- ctrl-c on terminal, both child process, and parent process should get interupted. 
- kill SIGINT/SIGTERM directly to parent voyager process; Again, both parent process and child processes should exit. 
- kill SIGINT/SIGTERM to child process. Parent process should not die, it should continue to the next assessment (depending on continue-on-error)

Reply:
Yeah it works as expected now. 
For the third one, it is anyways proceeding to the next assessment, since there is no error as such.

Reply:
>For the third one, it is anyways proceeding to the next assessment, since there is no error as such. 

Oh, but ideally we should detect, and report that error. Would be wrong to silently continue (especially if someone passed --continue-on-error false)

Reply:
The challenge here for to exit as error, the parent process is not getting any error from the child function call. so it assumes that everything worked fine.

Reply:
I believe there was some discussion in the standup regarding this one to send exit code 1  from child when interrupted to parent, are we going to take that also in separate PR?

Reply:
handled for now with the help of sleep. 
Separate ticket for proper handling the issue - https://yugabyte.atlassian.net/issues/DB-12388